include/rocksdb/memtablerep.h

```diff
@@ -194,6 +196,15 @@ class MemTableRep {
     return 0;
   }
 
+  // Returns a vector of unique random memtable entries of size 'sample_size'.
+  virtual void UniqueRandomSample(const uint64_t& num_entries,
+                                  const uint64_t& sample_size,
+                                  std::unordered_set<const char*>* entries) {
+    (void)num_entries;
```

Comment:
Seems like there should be an assert(false) or have the caller explicitly check for no entries. The declaration in MemTable should document what an implementation that doesn't support the operation should do.